src/hotspot/share/oops/symbolHandle.hpp

```diff
@@ -53,6 +77,13 @@ class SymbolHandleBase : public StackObj {
   SymbolHandleBase(Symbol *s) : _temp(s) {
     if (!TEMP) {
       Symbol::maybe_increment_refcount(_temp);
+      return;
+    }
+
+    // Delay cleanup for temp symbols. But don't requeue existing entries,
+    // or entries that are held elsewhere - it's a waste of effort.
+    if (s != nullptr && s->refcount() == 1) {
+      add_to_cleanup_delay_queue(s);
```

Comment:
This could add the comment that adding to the delay queue will increment the refcount again for the entry while in the queue (had to look again to verify that).